### PR TITLE
Re-enable `5.6` for Elastic products

### PR DESCRIPTION
**Pull request by dliappis**
Migrating existing users of the library images to the Elastic provided
ones during a minor release (`5.6`) is a big, potentially breaking,
change. It seems like a safer approach to keep producing images until 
`6.0.0` is out, so that the community can benefit from bug and security fixes 
throughout 5.x and give more time to adjust to the new images.

Extend support for library images through `5.6.x` until `6.0.0` has
been released. Also update the product logos.

Relates #945

### CODE REVIEW

**Comment by tianon** [comment]
Seems sane -- do we need to also cc @jasontedor and/or @elasticdog? :innocent:

(This is of course just a small portion the work to be done to bring these back now -- we've also got to un-archive the repos and re-enable their `update.sh` automation, pulling back in `library/elasticsearch`, `library/kibana`, and `library/logstash` over in https://github.com/docker-library/official-images)

**Comment by tianon** [comment]
(Threading the needle: https://github.com/docker-library/docs/pull/842 is a previously related PR, and the discussion in https://github.com/docker-library/elasticsearch/pull/160 is highly relevant, and was the catalyst for the deprecation in the first place.)

File: elasticsearch/deprecated.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
-This image has been deprecated in favor of the [official `elasticsearch` image](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html) provided and maintained by [elastic.co](https://www.elastic.co/). The upstream images are available to pull via `docker.elastic.co/elasticsearch/elasticsearch:[version]` like `5.4.2`. The images found here will receive no further updates once the `5.6.0` release is available upstream. Please adjust your usage accordingly.
+This image has been deprecated in favor of the [official `elasticsearch` image](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html) provided and maintained by [elastic.co](https://www.elastic.co/). The list of images available from Elastic can be found at www.docker.elastic.co.  The images found here will receive no further updates once the `6.0.0` release is available upstream. Please adjust your usage accordingly.
```
Comment (tianon) [inline elasticsearch/deprecated.md]: This `www.docker.elastic.co` reference ought to be a link, ala:
```markdown
... can be found at [www.docker.elastic.co](https://www.docker.elastic.co/). ...
```

**Comment by tianon** [comment]
Also, you don't happen to know a more specific timeline for 6.0, right? (it's helpful to users to note the exact cut-off date, but if it's not possible to know yet, that's obviously fair :smile:)

**Comment by elasticdog** [comment]
Yep, this is all good with me... @dliappis is just pushing forward with the plan that we've discussed internally here at Elastic. I believe @skearns64, our Sr. Director of Product Management has also been in contact with someone internally there at Docker regarding the situation (although I'm not sure whom in particular). Assistance on re-enabling the `update.sh` automation work would be greatly appreciated, if it's not something we can easily PR for you.

**Comment by tianon** [comment]
Ok, cool; just double-checking! :+1: :heart: (especially since this is the first / only we've heard of it :smile:)

I'm happy to go ahead and take over from here; thanks! :+1: (I think we actually need to get the image added back in https://github.com/docker-library/official-images/tree/master/library first, so I'll start there and get the repos de-archived, etc and come back to merge this once all that's in place)

**Comment by tianon** [comment]
Alllright, got Travis green here, and got the `update.sh` infrastructure back up:

- https://github.com/docker-library/elasticsearch/commit/e3b9b5c27ce4ea5fbdbfac57280a98e54bf7528f
- https://github.com/docker-library/logstash/commit/c257f0d648cd01631657a39e005911fc4c7c5aeb
- https://github.com/docker-library/kibana/commit/8a791a6db074dec988aad1e1f6ac645b50ee6702

And the Official Images PR to officially restore the images as well: https://github.com/docker-library/official-images/pull/3610

**Pull request merged**